dali/pipeline/operators/color/brightness_contrast.h

```diff
@@ -0,0 +1,193 @@
+struct Kernel<GPUBackend, Out, In, ndims> {
+  using type = kernels::brightness_contrast::BrightnessContrastGpu<Out, In, ndims>;
+};
+
+
+template <class Backend>
+struct ArgumentType {
+  static_assert(
+          std::is_same<Backend, CPUBackend>::value || std::is_same<Backend, GPUBackend>::value,
+          "Unsupported Backend");
+  using type = float;
+};
+
+
+template <>
+struct ArgumentType<GPUBackend> {
+  using type = std::vector<float>;
+};
+
+
+/**
+ * Select proper type for argument (for either sample processing or batch processing cases)
+ */
+template <class Backend>
+using argument_t = typename ArgumentType<Backend>::type;
+
+
+/**
+ * Chooses proper kernel (CPU or GPU) for given template parameters
+ */
+template <class Backend, class OutputType, class InputType, size_t ndims>
+using BrightnessContrastKernel = typename Kernel<Backend, OutputType, InputType, ndims>::type;
+
+
+/**
+ * Assign argument, whether it is a single value (for sample-wise processing)
+ * or vector of values (for batch processing, where every argument is defined per-sample)
+ */
+template <class Backend>
+void assign_argument_value(const OpSpec &, const std::string &, argument_t<Backend> &) {
+  DALI_FAIL("Unsupported Backend. You may want to write your own specialization.");
+}
+
+
+template <>
+void assign_argument_value<CPUBackend>(const OpSpec &spec, const std::string &arg_name,
+                                       argument_t<CPUBackend> &arg) {
+  std::vector<float> tmp;
+  GetSingleOrRepeatedArg(spec, tmp, arg_name);
+  arg = tmp[0];
+}
+
+
+template <>
+void assign_argument_value<GPUBackend>(const OpSpec &spec, const std::string &arg_name,
+                                       argument_t<GPUBackend> &arg) {
+  GetSingleOrRepeatedArg(spec, arg, arg_name);
+}
+
+}  // namespace detail
+
+
+template <class Backend>
+class BrightnessContrast : public Operator<Backend> {
+ public:
+  explicit BrightnessContrast(const OpSpec &spec) :
+          Operator<Backend>(spec),
+          output_type_(spec.GetArgument<DALIDataType>(detail::kOutputType)) {
+    detail::assign_argument_value<Backend>(spec, detail::kBrightness, brightness_);
+    detail::assign_argument_value<Backend>(spec, detail::kContrast, contrast_);
+    if (std::is_same<Backend, GPUBackend>::value) {
+      kernel_manager_.Resize(1, 1);
+    } else {
+      kernel_manager_.Resize(num_threads_, batch_size_);
+    }
+  }
+
+
+  ~BrightnessContrast() = default;
+
+
+  DISABLE_COPY_MOVE_ASSIGN(BrightnessContrast);
+
+
+ protected:
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+
+  bool SetupImpl(std::vector<::dali::OutputDesc> &output_desc,
+                 const ::dali::workspace_t<Backend> &ws) override {
+    const auto &input = ws.template InputRef<Backend>(0);
+    const auto &output = ws.template OutputRef<Backend>(0);
+    output_desc.resize(1);
+
+    TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float), (
```

Comment:
indentation is not consistent in this file. e.g mixed 4-space and 2-space indents

Reply:
I decided on a little bigger indentation because this `TYPE_SWITCH(TYPE_SWITCH...)` thing gets clearer this way